sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/StreamingJoinHelper.scala

```diff
@@ -96,17 +96,22 @@ object StreamingJoinHelper extends PredicateHelper with Logging {
       // If state watermark is W, all state where timestamp <= W will be cleaned up.
       // Now when the canonicalized join condition solves to leftTime >= W, we don't want to clean
       // up leftTime <= W. Rather we should clean up leftTime <= W - 1. Hence the -1 below.
-      val stateWatermark = predicate match {
-        case LessThan(l, r) => getStateWatermarkSafely(l, r)
-        case LessThanOrEqual(l, r) => getStateWatermarkSafely(l, r).map(_ - 1)
-        case GreaterThan(l, r) => getStateWatermarkSafely(r, l)
-        case GreaterThanOrEqual(l, r) => getStateWatermarkSafely(r, l).map(_ - 1)
-        case _ => None
-      }
-      if (stateWatermark.nonEmpty) {
-        logInfo(s"Condition $joinCondition generated watermark constraint = ${stateWatermark.get}")
-      }
-      stateWatermark
+      val stateWatermarks = (predicate match {
```

Comment:
why do we change the streaming code?

Reply:
I got failures in streaming tests. It turns out that they calculate upper/lower bounds based on `if x > y` conditions. With `Between` op we no longer have expressions they can use so I had to extend the logic to cover `Between` for this watermark resolution.

Reply:
I think we can keep the code structure unchanged, with one more patter match
```
case Between(p, lower, upper, _) =>
  getStateWatermarkSafely(lower, p).map(_ - 1).orElse(getStateWatermarkSafely(p, upper).map(_ - 1))
```